components/MainMenu/BurgerIcon.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { BurgerIconProps } from "./types/BurgerIconProps";
+const BurgerIcon = (props: BurgerIconProps) => {
+  const { styles, customIcon, barClassName, className, onClick, onIconHoverChange } = props;
+  const [hover, setHover] = useState(false);
+  const getLineStyle = (index: number) => {
+    return {
+      position: "absolute" as "absolute",
+      height: "20%",
+      left: 0,
+      right: 0,
+      top: 20 * (index * 2) + "%",
+      opacity: hover ? 0.6 : 1,
+      ...(hover && styles?.bmBurgerBarsHover)
+    };
+  };
+  let icon;
+  let buttonStyle = {
```

Comment:
It looks like we have some let declarations here that are never reassigned, it would be better to use const for these variables